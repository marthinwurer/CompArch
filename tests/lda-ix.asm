; test indirect indexed addressing with lda/ldb
	.entry	start
;
	.org	0x50
loc1:	.word	0
loc2:	.word	1
loc3:	.word	2
loc4:	.word	3
;
	.org	0x60
loc5:	.word	0xfffff
loc6:	.word	0xffffe
loc7:	.word	0xffffd
loc8:	.word	0xffffc
;
	.org	0x100
arr1:	.word	0x10
	.word	0x20
	.word	0x30
	.word	0x40
;
	.org	0x110
arr2:	.word	0x100
	.word	0x200
	.word	0x300
	.word	0x400
;
	.org	0x150
ptr1:	.word	0x100
ptr2:	.word	0x114
;
	.org	0x200
start:
	ldxr	ptr1
	lda	loc1,ix
	lda	loc2,ix
	lda	loc3,ix
	lda	loc4,ix
;
	ldxr	ptr2
	ldb	loc5,ix
	ldb	loc6,ix
	ldb	loc7,ix
	ldb	loc8,ix
;
	halt
