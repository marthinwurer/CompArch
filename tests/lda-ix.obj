50 1 00000
51 1 00001
52 1 00002
53 1 00003
60 1 fffff
61 1 ffffe
62 1 ffffd
63 1 ffffc
100 1 00010
101 1 00020
102 1 00030
103 1 00040
110 1 00100
111 1 00200
112 1 00300
113 1 00400
150 1 00100
151 1 00114
200 1 18150
201 1 d0050
202 1 d0051
203 1 d0052
204 1 d0053
205 1 18151
206 1 d1060
207 1 d1061
208 1 d1062
209 1 d1063
20a 1 00000
200
