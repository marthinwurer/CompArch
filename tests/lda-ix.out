CPU "ARCH" Simulator, 2.5a(Jan 22 2015)
-----------------------------------------

Memory sets starting address to 200
200:  18150  LDXR  150  A[00000]  B[00000]  XR[000]
201:  d0050  LDA   100  A[00000]  B[00000]  XR[100]
202:  d0051  LDA   101  A[00010]  B[00000]  XR[100]
203:  d0052  LDA   102  A[00020]  B[00000]  XR[100]
204:  d0053  LDA   103  A[00030]  B[00000]  XR[100]
205:  18151  LDXR  151  A[00040]  B[00000]  XR[100]
206:  d1060  LDB   113  A[00040]  B[00000]  XR[114]
207:  d1061  LDB   112  A[00040]  B[00400]  XR[114]
208:  d1062  LDB   111  A[00040]  B[00300]  XR[114]
209:  d1063  LDB   110  A[00040]  B[00200]  XR[114]
20a:  00000  HALT       A[00040]  B[00100]  XR[114]

Machine Halted - HALT instruction executed

Simulated time 80 cycles

LAST CPUObject DESTROYED; END OF SIMULATION
